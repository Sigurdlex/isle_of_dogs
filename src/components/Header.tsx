import React, { ReactNode } from 'react';
import styled from 'styled-components';

import { ROYALBLUE } from 'constants/colors';

export const Header = ({ children }: { children: ReactNode }) => (
  <ContainerHeader>
    <Content>
      {children}
    </Content>
  </ContainerHeader>
);

const ContainerHeader = styled.header`
  min-width: 700px;
  z-index: 2;
  position: fixed;
  background: ${ROYALBLUE};
  height: 100px;
  width: 100%;
  top: 0;
  display: flex;
  box-shadow: 0 1px 15px black;
`;

const Content = styled.div`
  width: 80%;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
