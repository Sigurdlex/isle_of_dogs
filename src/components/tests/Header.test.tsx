import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { Header } from '../Header';

afterEach(cleanup);

const CHILDREN = 'CHILDREN';

test('<Header />', () => {
  const { container } = render(<Header>{CHILDREN}</Header>);
  const element: HTMLElement = container.firstChild as HTMLElement;
  expect(element.textContent).toMatch(CHILDREN);
  expect(element).toMatchSnapshot();
});
