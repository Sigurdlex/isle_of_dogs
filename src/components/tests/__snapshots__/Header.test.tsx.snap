// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 1`] = `
.c0 {
  min-width: 700px;
  z-index: 2;
  position: fixed;
  background: #3d3afd;
  height: 100px;
  width: 100%;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 1px 15px black;
}

.c1 {
  width: 80%;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    CHILDREN
  </div>
</header>
`;
